tests/conftest.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import pytest
+import uuid
+
+from common_ci_utils.command_runner import exec_cmd
+from framework.connection import SSHConnection
+from noobaa_sa.factories import AccountFactory
+
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def account_manager(account_json=None):
+    account_factory = AccountFactory()
+    return account_factory.get_account(account_json)
+
+
+@pytest.fixture
+def unique_resource_name(request):
+    """
+    Generates a unique resource name with the given prefix
+    """
+
+    def _get_unique_name(prefix="resource"):
+        unique_id = str(uuid.uuid4()).split("-")[0]
+        return f"{prefix}_{unique_id}"
+
+    return _get_unique_name
+
+
+@pytest.fixture(scope="function")
+def random_hex(request):
+    """
+    Generates a random hexadecimal string.
+    """
+
+    def _get_random_hex():
+        cmd = "openssl rand -hex 20"
+        completed_process = exec_cmd(cmd)
+        stdout = completed_process.stdout
+        return stdout.decode("utf-8").strip()
+
+    return _get_random_hex
+
+
+@pytest.fixture(scope="session")
+def conn(request):
+    """
+    Generates a random hexadecimal string.
+    """
+    ssh_conn = SSHConnection()
+    # store the conn instance in config
+    request.config._conn_fixture = ssh_conn
+    connection = ssh_conn.get_connection()
+    return connection
+
+
+def pytest_unconfigure(config):
+    # get the conn instance from the session and close it
+    ssh_conn = config._conn_fixture
+    if ssh_conn:
+        ssh_conn.close_connection()
```

Comment:
If we want to make sure the connection is closed, then we should just access the object from here:
SSHConnection().close_connection() 

Handling if connection exist should be handled from function itself.

Reply:
removed this and handled in pytest_sessionfinish